move-mutation-test/tests/testsuite.rs

```diff
@@ -0,0 +1,73 @@
+use aptos::common::types::MovePackageDir;
+use log::info;
+use move_mutation_test::cli::CLIOptions;
+use move_mutation_test::cli::TestBuildConfig;
+use move_mutation_test::run_mutation_test;
+use mutator_common::report::Report;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+const RED_ZONE: usize = 128 * 1024; // 128 KiB
+const STACK_SIZE: usize = 32 * RED_ZONE; // 4 MiB
+
+fn test_run_mutation_test(path: &Path, expected_report: String) -> datatest_stable::Result<()> {
+    let expected_report =
+        Report::load_from_str(expected_report).expect("failed to load the report");
+    let package_path = path.parent().expect("package path not found");
+
+    let mut move_pkg = MovePackageDir::new();
+    move_pkg.package_dir = Some(PathBuf::from(package_path));
+    let test_build_cfg = TestBuildConfig {
+        move_pkg,
+        dump_state: false,
+        filter: None,
+        ignore_compile_warnings: false,
+        apply_coverage: true,
+        gas_limit: 1_000,
+    };
+
+    let report_file = PathBuf::from("report.txt");
+    let cli_opts = CLIOptions {
+        output: Some(report_file.clone()),
+        ..Default::default()
+    };
+
+    stacker::maybe_grow(RED_ZONE, STACK_SIZE, || {
+        run_mutation_test(&cli_opts, &test_build_cfg).expect("running the mutation test failed");
+        info!(
+            "remaining stack size is {}",
+            stacker::remaining_stack().expect("failed to get the remaining stack size")
+        );
+    });
+
+    let generated_report = Report::load_from_json_file(&report_file).expect("report not found");
+
+    // Let's make sure the reports are equal.
+    let Report {
+        files: mut expected_entries,
+        ..
+    } = expected_report;
+    let Report {
+        files: generated_report_files,
+        ..
+    } = generated_report;
+
+    for (file, mutant_stats) in generated_report_files {
```

Comment:
`We can't compare files directly, since the `package_path` would be updated the with absolute path:

```json
          "--- original\n+++ modified\n@@ -1,6 +1,6 @@\n module TestAccount::Sum {\n     fun sum(x: u128, y: u128): u128 {\n-        let sum_r = x + y;\n+        let sum_r = x - y;\n         spec {\n                 assert sum_r == x+y;\n         };\n",
          "--- original\n+++ modified\n@@ -1,6 +1,6 @@\n module TestAccount::Sum {\n     fun sum(x: u128, y: u128): u128 {\n-        let sum_r = x + y;\n+        let sum_r = x * y;\n         spec {\n                 assert sum_r == x+y;\n         };\n",
          "--- original\n+++ modified\n@@ -1,6 +1,6 @@\n module TestAccount::Sum {\n     fun sum(x: u128, y: u128): u128 {\n-        let sum_r = x + y;\n+        let sum_r = x / y;\n         spec {\n                 assert sum_r == x+y;\n         };\n",
          "--- original\n+++ modified\n@@ -1,6 +1,6 @@\n module TestAccount::Sum {\n     fun sum(x: u128, y: u128): u128 {\n-        let sum_r = x + y;\n+        let sum_r = x % y;\n         spec {\n                 assert sum_r == x+y;\n         };\n"
        ]
      }
    ]
  },
  "package_dir": "move-spec-testing/move-mutator/tests/move-assets/simple"
}
```

In the examples above (I included in the PR), I removed my `home` path info, but that's what would make two reports different on two different machines

Reply:
Added a comment so it's obvios from the test code as well.